asserts/asserts.go

```diff
@@ -193,17 +193,6 @@ func Decode(serializedAssertion []byte) (Assertion, error) {
 	return buildAssertion(headers, body, content, signature)
 }
 
-func checkMandatory(headers map[string]string, name string) (string, error) {
-	value, ok := headers[name]
-	if !ok {
-		return "", fmt.Errorf("assertion %v header is mandatory", name)
-	}
-	if len(value) == 0 {
-		return "", fmt.Errorf("assertion %v should not be empty", name)
-	}
-	return value, nil
-}
-
 func checkInteger(headers map[string]string, name string) (int, error) {
```

Comment:
I can move checkAssertType, checkInteger has the defaulting to 0 behavior, would like a 3rd use case before moving it, it may need more params to control that

checkRevision should only be used here (asserts.go)